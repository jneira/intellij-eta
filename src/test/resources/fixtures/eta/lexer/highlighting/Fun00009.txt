ITinline_prag ('{-#')
ITline_prag (' LANGUAGE TypeOperators ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE MagicHash ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00009')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Digit')
WHITE_SPACE (' ')
ITvarid ('a')
ITcparen (')')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n       ')
ITinline_prag ('{-#')
ITline_prag (' SPECIALIZE instance Sized (Digit (Elem a)) ')
ITline_prag ('#-}')
WHITE_SPACE ('\n       ')
ITinline_prag ('{-#')
ITline_prag (' SPECIALIZE instance Sized (Digit (Node a)) ')
ITline_prag ('#-}')
WHITE_SPACE ('\n       ')
ITvarid ('size')
WHITE_SPACE (' ')
ITvarid ('xs')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('foldl')
WHITE_SPACE (' ')
IToparen ('(')
ITvarsym ('\')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITvarsym ('+')
WHITE_SPACE (' ')
ITvarid ('size')
WHITE_SPACE (' ')
ITvarid ('x')
ITcparen (')')
WHITE_SPACE (' ')
ITinteger ('0')
WHITE_SPACE (' ')
ITvarid ('xs')
WHITE_SPACE ('\n\n')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n       ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITvbar ('|')
WHITE_SPACE (' ')
ITvarid ('null')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITrational ('9.3')
WHITE_SPACE ('\n       ')
ITconid ('Nothing')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITrational ('7.1')
WHITE_SPACE ('\n\n')
ITdata ('data')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('a1')
WHITE_SPACE (' ')
ITconsym (':<')
WHITE_SPACE (' ')
ITvarid ('a2')
ITcparen (')')
WHITE_SPACE ('  ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Foo')
WHITE_SPACE ('\n\n\n')
ITvarid ('f1')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITlet ('let')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Float')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('x')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITrational ('10.2#')
WHITE_SPACE ('\n         ')
ITvarid ('y')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Double')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('y')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITrational ('10.7##')
WHITE_SPACE ('\n         ')
ITvarid ('z')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Int')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('z')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('1#')
WHITE_SPACE ('\n         ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Word')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('a')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('9##')
WHITE_SPACE ('\n         ')
ITvarid ('b')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Char')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('b')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITchar (''B'#')
WHITE_SPACE ('\n         ')
ITvarid ('c')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('String')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITvarid ('c')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITstring ('"')
ITstring ('Hello, world!')
ITstring ('"')
ITvarsym ('#')
WHITE_SPACE ('\n     ')
ITin ('in')
WHITE_SPACE (' ')
ITinteger ('19')
