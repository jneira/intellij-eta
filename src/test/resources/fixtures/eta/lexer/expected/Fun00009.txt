ITblockComment ('{-# LANGUAGE TypeOperators #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE MagicHash #-}')
WHITE_SPACE ('\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00009')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITvocurly ('')
WHITE_SPACE ('')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Digit')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n       ')
ITvocurly ('')
WHITE_SPACE ('')
ITspec_prag ('{-# SPECIALIZE')
WHITE_SPACE (' ')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Digit')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Elem')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITclose_prag ('#-}')
WHITE_SPACE ('\n       ')
ITsemi ('')
WHITE_SPACE ('')
ITspec_prag ('{-# SPECIALIZE')
WHITE_SPACE (' ')
ITinstance ('instance')
WHITE_SPACE (' ')
ITconid ('Sized')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Digit')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Node')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITclose_prag ('#-}')
WHITE_SPACE ('\n       ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('size')
WHITE_SPACE (' ')
ITvarid ('xs')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('foldl')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITlam ('\')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('i')
WHITE_SPACE (' ')
ITvarsym ('+')
WHITE_SPACE (' ')
ITvarid ('size')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITinteger ('0')
WHITE_SPACE (' ')
ITvarid ('xs')
WHITE_SPACE ('\n\n')
ITvccurly ('')
WHITE_SPACE ('')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('f')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITcase ('case')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITof ('of')
WHITE_SPACE ('\n       ')
ITvocurly ('')
WHITE_SPACE ('')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITvbar ('|')
WHITE_SPACE (' ')
ITvarid ('null')
WHITE_SPACE (' ')
ITvarid ('x')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITrational ('9.3')
WHITE_SPACE ('\n       ')
ITsemi ('')
WHITE_SPACE ('')
ITconid ('Nothing')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITrational ('7.1')
WHITE_SPACE ('\n\n')
ITvccurly ('')
WHITE_SPACE ('')
ITsemi ('')
WHITE_SPACE ('')
ITdata ('data')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('a1')
WHITE_SPACE (' ')
ITconsym (':<')
WHITE_SPACE (' ')
ITvarid ('a2')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('  ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Foo')
WHITE_SPACE ('\n\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('f1')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITlet ('let')
WHITE_SPACE (' ')
ITvocurly ('')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Float')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('x')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITrational ('10.2')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('y')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Double')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('y')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITrational ('10.7')
WHITE_SPACE ('')
ITvarsym ('##')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('z')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Int')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('z')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('1')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Word')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('a')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITinteger ('9')
WHITE_SPACE ('')
ITvarsym ('##')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('b')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Char')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('b')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITchar (''B'')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('c')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('String')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n         ')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('c')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITstring ('"Hello, world!"')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE ('\n     ')
ITvccurly ('')
WHITE_SPACE ('')
ITin ('in')
WHITE_SPACE (' ')
ITinteger ('19')